src/backend/distributed/transaction/transaction_management.c

```diff
@@ -459,12 +525,7 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 		case XACT_EVENT_PARALLEL_PRE_COMMIT:
 		case XACT_EVENT_PRE_PREPARE:
 		{
-			if (InCoordinatedTransaction())
-			{
-				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-								errmsg("cannot use 2PC in transactions involving "
-									   "multiple servers")));
-			}
+			EnsureSafe2PC();
```

Comment:
Maybe: EnsurePrepareTransactionIsAllowed();

This code primarily tries to forbid the user from sending a PREPARE TRANSACTION after writing to a distributed table, which would be unrelated to Citus doing a 2PC.

Reply:
Changed.